### PR TITLE
Homework 4

### PR DESCRIPTION
**Pull request by Julie8899**

### CODE REVIEW

File: HW4/HW4.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+# from io import StringIO
+
+# from sklearn import datasets, tree
+# from sklearn.metrics import confusion_matrix
+# from sklearn.model_selection import GridSearchCV
+# from sklearn.tree import DecisionTreeClassifier, export_graphviz
+
+# import tempfile
+# from PIL import Image
+
```
Comment (huongphamBDA) [inline HW4/HW4.py]: You need to use the `def main()` function and `if __name__ == "__main__":`

File: HW4/HW4.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+data_df = pandas.read_csv("suspectheartrates.csv")
+
+
+def is_column_cat(df, column_name):
```
Comment (huongphamBDA) [inline HW4/HW4.py:19]: I thought we need to say out loud whether the response is continuous or boolean, and whether each predictor is cat/cont. Do you mean if df[column_name].nunique() > 2, then that column is continuous?

File: HW4/HW4.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            # Pred is cont and Res is cont, plot scatter with trendline
+            # Calculate p-value and t-score, Linear Regression
+            plot_scatter(data_df, pred_col_name, res_col_name)
+            plot_mean_sq_diff(
+                data_df, pred_col_name, res_col_name, weighted=False, n_bins=15
+            )
+            plot_mean_sq_diff(
+                data_df, pred_col_name, res_col_name, weighted=True, n_bins=30
+            )
+            pass
```
Comment (huongphamBDA) [inline HW4/HW4.py]: I don't see you doing the Random Forest part, calculating p t values, and generating a table with variables and rankings

File: HW4/HW4.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        yaxis_title="Distribution",
+    )
+    fig.show()
+    return
+
+
+def plot_scatter(df, pred_col_name, res_col_name):
+    pred_values = df[pred_col_name].tolist()
+    res_values = df[res_col_name].tolist()
+
```
Comment (huongphamBDA) [inline HW4/HW4.py:64]: How about Logistic Regression when the response is boolean type? I think boolean is when the column has only two values (True or False; 0 or 1). If it has two random values other than 0 and 1, it's categorical and not boolean.

File: HW4/HW4.py (hunk 1 above)
Comment (Julie8899) [reply HW4/HW4.py]: I updated it. Thanks!

File: HW4/HW4.py (hunk 2 above)
Comment (Julie8899) [reply HW4/HW4.py:19]: Yes.

File: HW4/HW4.py (hunk 3 above)
Comment (Julie8899) [reply HW4/HW4.py]: I just commit a new version but I have not generated a table with variables and ranking.

**Pull request merged**